polynote-kernel/src/main/scala/polynote/kernel/lang/scal/ScalaSource.scala

```diff
@@ -1,29 +1,33 @@
 package polynote.kernel.lang.scal
 
 import cats.data.Ior
+import cats.effect.IO
 import cats.syntax.either._
 import polynote.kernel.EmptyCell
-import polynote.kernel.util.KernelReporter
+import polynote.kernel.lang.LanguageInterpreter
+import polynote.kernel.util.{CellContext, KernelContext, KernelReporter}
 import polynote.messages.CellID
 
 import scala.annotation.tailrec
 import scala.collection.immutable.{ListMap, Queue}
 import scala.reflect.internal.util.{ListOfNil, Position, RangePosition, SourceFile}
+import scala.tools.nsc.interactive.Global
 
 /**
   * Captures some Scala source in the context of a notebook, and provides compilation operations.
   *
   * TODO: the refined Interpreter type is really annoying; dictated by Global's path-dependent types. Can that be
   *       unwound?
   */
-class ScalaSource[Interpreter <: ScalaInterpreter](val interpreter: Interpreter)(
-  id: CellID,
-  availableSymbols: Set[Interpreter#Decl],
-  previousSources: List[ScalaSource[Interpreter]],
-  code: String
+class ScalaSource[G <: Global](
```

Comment:
I tried, I really tried to get rid of this type bound thing. But it just doesn't work. I was able to get it to `Global`, at least, and you can generally avoid errors if you import the right `Global` (even though they're all the same)

Reply:
Bummer!